org/mozilla/jss/ssl/SSLCipher.c

```diff
@@ -0,0 +1,31 @@
+#include <nspr.h>
+#include <nss.h>
+#include <ssl.h>
+#include <pk11pub.h>
+#include <jni.h>
+
+#include "_jni/org_mozilla_jss_ssl_SSLCipher.h"
+
+JNIEXPORT jboolean JNICALL
+Java_org_mozilla_jss_ssl_SSLCipher_checkSupportedStatus(JNIEnv *env, jclass clazz, jint cipher_suite)
+{
+    SSLCipherSuiteInfo info = { 0 };
+    jboolean found = JNI_FALSE;
+
+    for (PRUint16 index = 0; index < SSL_NumImplementedCiphers; index++) {
+        if (SSL_ImplementedCiphers[index] == cipher_suite) {
+            found = JNI_TRUE;
+            break;
+        }
+    }
+
+    if (found == JNI_FALSE) {
+        return found;
+    }
+
+    if (SSL_GetCipherSuiteInfo(cipher_suite, &info, sizeof(info)) != SECSuccess || info.length < sizeof(info)) {
+        return JNI_FALSE;
+    }
+
+    return ((!PK11_IsFIPS() || (info.isFIPS == 0)) && (info.nonStandard == 0)) ? JNI_TRUE : JNI_FALSE;
```

Comment:
Shouldn't that be `info.isFIPS != 0`?

This expression is also a bit hard to read. Could we break it down maybe like this?
```
if (info.nonStandard != 0) {
    // non-standard ciphers are not supported
    return JNI_FALSE;
}
if (info.isFIPS == 0 && PKI_IsFIPS()) {
    // non-FIPS ciphers are not supported in FIPS mode
    return JNI_FALSE;
}
return JNI_TRUE;
```